migrations/receipts.go

```diff
@@ -18,57 +19,77 @@ import (
 
 var receiptsCborEncode = Migration{
 	Name: "receipts_cbor_encode",
-	Up: func(db ethdb.Database, datadir string, OnLoadCommit etl.LoadCommitHandler) error {
+	Up: func(db ethdb.Database, datadir string, progress []byte, CommitProgress etl.LoadCommitHandler) error {
 		logEvery := time.NewTicker(30 * time.Second)
 		defer logEvery.Stop()
+		buf := make([]byte, 0, 100_000)
+		const loadStep = "load"
+
 		collector, err1 := etl.NewCollectorFromFiles(datadir)
 		if err1 != nil {
 			return err1
 		}
-		if collector == nil {
-			collector = etl.NewCriticalCollector(datadir, etl.NewSortableBuffer(etl.BufferOptimalSize))
-			buf := make([]byte, 0, 100_000)
-			if err1 = db.Walk(dbutils.BlockReceiptsPrefix, nil, 0, func(k, v []byte) (bool, error) {
-				blockNum := binary.BigEndian.Uint64(k[:8])
-				select {
-				default:
-				case <-logEvery.C:
-					var m runtime.MemStats
-					runtime.ReadMemStats(&m)
-					log.Info("Migration progress", "blockNum", blockNum, "alloc", common.StorageSize(m.Alloc), "sys", common.StorageSize(m.Sys))
+		switch string(progress) {
+		case "":
+			if collector != nil { //  can't use files if progress field not set
+				_ = os.RemoveAll(datadir)
+				collector, err1 = etl.NewCollectorFromFiles(datadir)
```

Comment:
What is the point of creating this collector if files are removed?

Reply:
Obviously, it will result in `collector == nil`, but we can just do it explicitely

Reply:
right, fixed